File: other/prime_checker.dart
import 'dart:io';
import 'dart:math';

/// 素数判定を行う関数
/// [n] 判定したい正の整数
/// 戻り値: 素数の場合true、そうでなければfalse
bool isPrime(int n) {
  if (n < 2) return false;
  if (n == 2) return true;
  if (n % 2 == 0) return false;
  
  // 3から√nまでの奇数で割り切れるかチェック
  int sqrtN = sqrt(n).toInt();
  for (int i = 3; i <= sqrtN; i += 2) {
    if (n % i == 0) return false;
  }
  return true;
}

/// 数値を処理する関数
/// [number] 処理する数値
void processNumber(int number) {
  if (number <= 0) {
    print('エラー: 正の整数を入力してください');
    return;
  }
  
  if (number > 10000000) {
    print('エラー: 10,000,000以下の数値を入力してください');
    return;
  }
  
  if (isPrime(number)) {
    print('$number は素数です');
  } else {
    print('$number は素数ではありません');
  }
}

void main(List<String> args) async {
  int? number;
  
  try {
    if (args.isNotEmpty) {
      // コマンドライン引数から取得
      number = int.parse(args[0]);
    } else {
      // ユーザー入力から取得
      stdout.write('判定したい数値を入力してください: ');
      String? input = stdin.readLineSync();
      if (input != null) {
        number = int.parse(input);
      }
    }
    
    if (number != null) {
      processNumber(number);
    } else {
      print('エラー: 有効な整数を入力してください');
    }
    
  } catch (e) {
    print('エラー: 有効な整数を入力してください');
  }
}

File: other/PrimeChecker.cs
using System;

/// <summary>
/// 素数判定プログラム（C#版）
/// </summary>
class PrimeChecker
{
    /// <summary>
    /// 素数判定を行うメソッド
    /// </summary>
    /// <param name="n">判定したい正の整数</param>
    /// <returns>素数の場合true、そうでなければfalse</returns>
    static bool IsPrime(int n)
    {
        if (n < 2) return false;
        if (n == 2) return true;
        if (n % 2 == 0) return false;

        // 3から√nまでの奇数で割り切れるかチェック
        for (int i = 3; i <= Math.Sqrt(n); i += 2)
        {
            if (n % i == 0) return false;
        }
        return true;
    }

    /// <summary>
    /// 数値を処理するメソッド
    /// </summary>
    /// <param name="number">処理する数値</param>
    static void ProcessNumber(int number)
    {
        if (number <= 0)
        {
            Console.WriteLine("エラー: 正の整数を入力してください");
            return;
        }

        if (number > 10000000)
        {
            Console.WriteLine("エラー: 10,000,000以下の数値を入力してください");
            return;
        }

        if (IsPrime(number))
        {
            Console.WriteLine($"{number} は素数です");
        }
        else
        {
            Console.WriteLine($"{number} は素数ではありません");
        }
    }

    static void Main(string[] args)
    {
        try
        {
            int number;

            if (args.Length > 0)
            {
                // コマンドライン引数から取得
                number = int.Parse(args[0]);
            }
            else
            {
                // ユーザー入力から取得
                Console.Write("判定したい数値を入力してください: ");
                string input = Console.ReadLine();
                number = int.Parse(input);
            }

            ProcessNumber(number);
        }
        catch (FormatException)
        {
            Console.WriteLine("エラー: 有効な整数を入力してください");
        }
        catch (OverflowException)
        {
            Console.WriteLine("エラー: 数値が範囲外です");
        }
        catch (Exception)
        {
            Console.WriteLine("エラー: 処理中に問題が発生しました");
        }
    }
}